test/LibraryManager.IntegrationTest/FileSaveRestoreTests.cs

```diff
@@ -0,0 +1,344 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using Microsoft.Test.Apex.VisualStudio;
+using Microsoft.Test.Apex.VisualStudio.Solution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Web.LibraryManager.IntegrationTest
+{
+    [TestClass]
+    public class FileSaveRestoreTests : VisualStudioLibmanHostTest
+    {
+        ProjectTestExtension _webProject;
+        ProjectItemTestExtension _libManConfig;
+        const string _projectName = @"TestProjectCore20";
+
+        [TestInitialize()]
+        public void initialize()
+        {
+            _webProject = Solution.ProjectsRecursive[_projectName];
+            _libManConfig = _webProject.Find(SolutionItemFind.FileName, "libman.json");
+        }
+
+        [TestCleanup]
+        public void Cleanup()
```

Comment:
This is for revert the libman.json file to its original status and also delete the libraries downloads during each test so as to not affect later tests.

Reply:
If you have to explain your code with a comment, then the code should be made more clear.  I think you could just add a comment `// Reset libman.json file contents`

You also assume here that the undo stack will always work/be accurate.  A better way to reset would be for TestInitialize to get the file content, and for Cleanup to set it back to the exact same content.

---
In reply to: [194349724](https://github.com/aspnet/LibraryManager/pull/113#discussion_r194349724) [](ancestors = 194349724)

Reply:
That make sense, I will try using this way to revert the change in the file.

---
In reply to: [194549775](https://github.com/aspnet/LibraryManager/pull/113#discussion_r194549775) [](ancestors = 194549775,194349724)